### ISSUE TITLE
Support for SFT / ChatDatasets

### ISSUE THREAD

**Issue by MaxiBoether** [eth-easl/mixtera#86]
Right now, Mixtera is built for pretraining. We don't have support for stuff like applying chat templates. Somewhat orthogonal but often related is supervised finetuning. SFT basically requires us to correctly mask the input and only train on the output.

https://github.com/huggingface/nanotron/pull/187/files

Here is an open PR to implement both in Nanotron. Probably fixing this will both require changes in Mixtera and https://github.com/MaxiBoether/nanotron-streaming
